### ISSUE TITLE
wish, set per-profile WM_CLASS  for X11 and app_id for Wayland

### ISSUE THREAD

**Issue by markstos** [pvsr/qpm#18]
For Qute instances to be seen as completely seperate apps by the windowing system, they should set a unique WM_CLASS under X11 and a custom app_id under Wayland. `qutebrowser` supports both currently in Git.

To set a custom WM_CLASS that will be recognized by X11, you can use: 

`qutebrowser --qt-arg name qutebrowser-profilename` 

This works in the published release. 

To set a custom app_id under Wayland is now possible in the Git version of `qutebrowser`. It hasn't been released yet but has been merged:

`qutebrowser --desktop-filename-name qutebrowser-profilename`

Related issue is: https://github.com/qutebrowser/qutebrowser/issues/5245

It should be safe to combine both `--qt-arg` and `--desktop-file-name`.

This issue is related to #17, which discusses generating custom `.desktop` files for profiles. It would be natural for these custom `.desktop` files to set unique WM_CLASS or `app_id` that matched their `.desktop` file name.

**Comment by markstos** [comment]
I'm testing the `--desktop-file-name` feature with `qutebrowser-profile` in my branch and it's working great:

https://github.com/jtyers/qutebrowser-profile/pull/12

However, working with `qutebrowser-profile` `bash` source code drives my nuts. Having a version in Python would a great alternative. 

For example, I created a "spotify" profile with `qutebrowser-profile`.  The Spotify desktop client uses X11 but sets WM_CLASS too late to match against with Sway. Spotify as a website doesn't necessarily include "Spotify" in the title, making matching by title hard.

But now by using Qutebrowser with --desktop-file-name 'qutebrowser-spotify' I can use the Sway option to match against `app_id` and automatically have the Spotify window to go a preferred workspace when it launches. I have a custom `.desktop` file for it so it's like a I have a custom Spotify client... powered by Qute!

**Comment by pvsr** [comment]
> I'm testing the `--desktop-file-name` feature with `qutebrowser-profile` in my branch and it's working great:
> 
> [jtyers/qutebrowser-profile#12](https://github.com/jtyers/qutebrowser-profile/pull/12)
> 
> However, working with `qutebrowser-profile` `bash` source code drives my nuts. Having a version in Python would a great alternative.

Interesting, I wasn't aware of that project. qpm started out as a shell script too but I rewrote it in python once I started wanting fancy stuff like control flow and string manipulation.

I added WM_CLASS support in https://github.com/pvsr/qpm/commit/11d7b79827ca00044446d9feb3ae27e17998eb5a and app_id with through a new `--set-app-id` flag in https://github.com/pvsr/qpm/commit/427fd121f9095a1da748f6899ee56dc3cf077d3c. Once `--desktop-file-name` has been released and out for a while maybe I'll make it the default.

Thanks for the suggestion!
